From 6a61ebf61b37b32dce5c221baf111acf21bb8916 Mon Sep 17 00:00:00 2001
From: Jaipal Singh <jpsinghgoud@gmail.com>
Date: Tue, 3 Jul 2018 13:04:42 +0530
Subject: [PATCH 14/47] Fixed minor issue of replacing

---
 examples/script.py | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/examples/script.py b/examples/script.py
index ba78997..57278fc 100644
--- a/examples/script.py
+++ b/examples/script.py
@@ -102,8 +102,7 @@ def calculate_value(text):
 	return float(result)
 
 def evaluate_expressions(commands):
-	print("ERE")
-	for command in commands:
+	for index, command in enumerate(commands):
 		if "exp" in command:
 			broken = command.split("[")
 			mystring = ""
@@ -112,6 +111,7 @@ def evaluate_expressions(commands):
 			and evaluate any expressions that might be present.
 			Because that's how expressions are meant to be written
 			'''
+			print('here')
 			for element in broken:
 				if "exp" in element:
 					'''
@@ -126,10 +126,9 @@ def evaluate_expressions(commands):
 					to_replace = "[" + element
 					print(to_replace)
 					command = command.replace(to_replace.strip(), str(result))
+					commands[index] = command
 					print(command)
 
-
-
 def create_proc_objects(proc_list):
 	for element in proc_list:
 		temp_list = element.split("\n")[0]
@@ -210,7 +209,6 @@ def parse_script(database_name, units, commands):
 	initalize_global_vars(commands)
 	initialize_procs(commands)
 	evaluate_expressions(commands)
-	exit()
 
 	for element in commands:
 		element = element.split()
-- 
2.14.3 (Apple Git-98)

