From 9f70e7738260be1f566e3a58756af281697f2ce0 Mon Sep 17 00:00:00 2001
From: Jaipal Singh <jpsinghgoud@gmail.com>
Date: Wed, 11 Jul 2018 16:02:23 +0530
Subject: [PATCH 21/47] Condensed to the core

---
 examples/sgi.py | 16 ++--------------
 1 file changed, 2 insertions(+), 14 deletions(-)

diff --git a/examples/sgi.py b/examples/sgi.py
index d976bf1..0c8985d 100644
--- a/examples/sgi.py
+++ b/examples/sgi.py
@@ -2,21 +2,15 @@
 This is the python version of sgi.sh using the python-brlcad module.
 
 usage:
-
     python sgi.py sgi.g
-
-    To Render : 
-
+To Render : 
     rtedge -s 1024 -F output.pix sgi.g cube.r
     pix-png -s 1024 < output.pix > output.png
 """
 
 import sys
-
 from brlcad.primitives import union
-
 from draw_primitive import *
-
 import brlcad.wdb as wdb
 
 def right(a, b, rcc_name, sph_name):
@@ -64,15 +58,11 @@ def down(a, b, rcc_name, sph_name):
 if __name__ == "__main__":
     argv = sys.argv
     database_name = argv[1]
-    
     # cube dimensions
     i, j, radius = 1000, 800, 100
-
     # starting position
     x, y, z = 0, 0, 0
-    
     brl_db = wdb.WDB(database_name, "SGI.g")
-
     rcc_name_list, sph_name_list, union_list = [], [], []
 
     for iterator in range (100, 118):
@@ -81,8 +71,6 @@ if __name__ == "__main__":
         sph_name_list.append(sph_name)
         union_list.extend((rcc_name, sph_name))
 
-    print(rcc_name_list)
-
     forward(100, i, rcc_name_list[0], sph_name_list[0])
     left(101, j, rcc_name_list[1], sph_name_list[1])
     down(102, i, rcc_name_list[2], sph_name_list[2])
@@ -108,4 +96,4 @@ if __name__ == "__main__":
             tree=union(union_list),
             shader="plastic {di=.8 sp=.2}",
             rgb_color=(64, 180, 96)
-        )
+        )
\ No newline at end of file
-- 
2.14.3 (Apple Git-98)

