From 1753c0b77fad00e496229c12ee48fb7e1edaaed9 Mon Sep 17 00:00:00 2001
From: Jaipal Singh <jpsinghgoud@gmail.com>
Date: Mon, 6 Aug 2018 20:42:03 +0530
Subject: [PATCH 45/47] Basic Shaders added for all primitives

---
 Tutorial/scene.py | 304 +++++++++++++++++++++++++++++++++++++++++++++++-------
 1 file changed, 266 insertions(+), 38 deletions(-)

diff --git a/Tutorial/scene.py b/Tutorial/scene.py
index 66dea79..99a3f85 100644
--- a/Tutorial/scene.py
+++ b/Tutorial/scene.py
@@ -10,19 +10,13 @@ if __name__ == "__main__":
             center=(0.5, 5, 8),
             radius=0.75
         )
+
         brl_db.rpp(
             "box1.s",
             pmin=(0, 0, 0),
             pmax=(2, 4, 2.5)
         )
-        brl_db.wedge(
-            "wedge1.s",
-            vertex=(0, 0, 3.5),
-            x_dir=(0, 1, 0),
-            z_dir=(0, 0, 1),
-            x_len=4, y_len=2, z_len=1,
-            x_top_len=3
-        )
+
         '''
         brl_db.arb4(
             "arb4.s",
@@ -223,32 +217,34 @@ if __name__ == "__main__":
             ]
         )
 
+        a = (88.7623, -18.0656, 0.601931)
+        b = (87.5712, -18.447, 4.56638)
+        c = (91.6581, -15.3077, 1.73722)
+        d = (91.4966, -21.1526, 1.12646)
+
         brl_db.bot(
             "bot.s",
             mode = 3,
             orientation = 1,
             flags = 0,
-            vertices = [(0, 5, 5), (0, 5, 6), (0, 6, 5), (1, 5, 5)],
+            vertices = [a, b, c, d],
             faces = [(0, 1, 2), (1, 2, 3), (3, 1, 0)],
             thickness=[2, 3, 1],
             face_mode=[True, True, False]
         )
-        
-        '''
 
-        brl_db.ebm(
-            "text1.s",
-            file_name = "ychar.bw",
-            x_dim = 10,
-            y_dim = 10,
-            tallness = 0.3)
-
-        brl_db.submodel(
-            "submodel.s",
-            file_name = "db.g",
-            treetop = "arb4.s",
+        brl_db.wedge(
+            "wedge1.s",
+            vertex=(87, -8.5, 0),
+            x_dir=(0, 0, 1),
+            z_dir=(0, -1, 0),
+            x_len=5, y_len=4, z_len=1,
+            x_top_len = 2
             )
 
+
+        
+        '''
         brl_db.sketch(
             name = "mysketch",
             base = (0, 0, 0),
@@ -256,24 +252,256 @@ if __name__ == "__main__":
             v_vec = (0, 1, 0),
             vertices = ((0,0), (1, 0), (0, 1), (1,1))
             )
-
         '''
 
-        '''
-        Error : ctypes.ArgumentError: argument 6: <type 'exceptions.TypeError'>: expected LP_c_double_Array_5 instance instead of LP_c_double_Array_5
+        brl_db.region(
+            'bot.r',
+            region_id = 1,
+            tree = 'bot.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
 
-        brl_db.metaball(
-            "metaball.s",      
-            points=[[(1, 1, 1), 1, 0], [(0, 0, 1), 2, 0]],
-            threshold=1, 
-            method=2, 
-        )
-        '''
+        brl_db.region(
+            'arb4.r',
+            region_id = 1,
+            tree = 'arb4.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'arb5.r',
+            region_id = 1,
+            tree = 'arb5.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'arb6.r',
+            region_id = 1,
+            tree = 'arb6.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'arb7.r',
+            region_id = 1,
+            tree = 'arb7.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'arb8.r',
+            region_id = 1,
+            tree = 'arb8.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'rhc.r',
+            region_id = 1,
+            tree = 'rhc2.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'rpc.r',
+            region_id = 1,
+            tree = 'rpc.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'epa.r',
+            region_id = 1,
+            tree = 'epa.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'ehy.r',
+            region_id = 1,
+            tree = 'ehy.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'particle.r',
+            region_id = 1,
+            tree = 'particle.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'rcc.r',
+            region_id = 1,
+            tree = 'rcc.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'tgc.r',
+            region_id = 1,
+            tree = 'tgc.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'tec.r',
+            region_id = 1,
+            tree = 'tec.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'rec.r',
+            region_id = 1,
+            tree = 'rec.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'trc.r',
+            region_id = 1,
+            tree = 'trc.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'sph.r',
+            region_id = 1,
+            tree = 'sph1.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'ellipsoid.r',
+            region_id = 1,
+            tree = 'ellipsoid.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'ellipsoid2.r',
+            region_id = 1,
+            tree = 'ellipsoid2.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+
+        brl_db.region(
+            'torus.r',
+            region_id = 1,
+            tree = 'torus.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'eto.r',
+            region_id = 1,
+            tree = 'eto.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'cone.r',
+            region_id = 1,
+            tree = 'cone.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'cone.r',
+            region_id = 1,
+            tree = 'cone.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'hyperboloid.r',
+            region_id = 1,
+            tree = 'hyperboloid.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'pipe.r',
+            region_id = 1,
+            tree = 'pipe.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+
+        brl_db.region(
+            'bot.r',
+            region_id = 1,
+            tree = 'bot.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+
+        brl_db.region(
+            'wedge.r',
+            region_id = 1,
+            tree = 'wedge1.s',
+            shader="plastic {di .8 sp .2}",
+            rgb_color=(255, 0, 96)
+            )
+        
+
+        brl_db.region(
+            name="all.r",
+            region_id = 123,
+            tree=(
+                "bot.s",
+                "sph1.s",
+                "ellipsoid2.s",
+                "wedge1.s",
+                "arb4.s",
+                "arb5.s",
+                "arb6.s",
+                "arb7.s",
+                "arb8.s",
+                "tec.s",
+                "rec.s",
+                "ellipsoid.s",
+                "torus.s",
+                "rcc.s",
+                "tgc.s",
+                "cone.s",
+                "trc.s",
+                "rpc.s",
+                "rhc2.s",
+                "epa.s",
+                "ehy.s",
+                "hyperboloid.s",
+                "eto.s",
+                "particle.s",
+                "pipe.s",
+            ),
 
-        '''
-        brl_db.half(
-            "half.s",
-            norm = (0, 0, 1), 
-            d = -1
         )
-        '''
-- 
2.14.3 (Apple Git-98)

