From cd3ed94705933600f2a94f9e7ccde2e2ddd69497 Mon Sep 17 00:00:00 2001
From: Jaipal Singh <jpsinghgoud@gmail.com>
Date: Tue, 7 Aug 2018 11:55:26 +0530
Subject: [PATCH 47/47] Minor updates

---
 Tutorial/core_tutorial.py | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Tutorial/core_tutorial.py b/Tutorial/core_tutorial.py
index a89f228..1eacec3 100644
--- a/Tutorial/core_tutorial.py
+++ b/Tutorial/core_tutorial.py
@@ -189,9 +189,6 @@ if __name__ == "__main__":
             face_mode=[True, True, False]
         )
 
-        '''
-        grip is created but unable to render
-        '''
         brl_db.grip(
             "grip.s",
             center = (0, 5, 1),
@@ -241,13 +238,11 @@ if __name__ == "__main__":
         )
         '''
 
-        '''
         brl_db.half(
             "half.s",
             norm = (0, 0, 1), 
             d = -1
         )
-        '''
 
         brl_db.region(
             name="all.r",
-- 
2.14.3 (Apple Git-98)

